### PR TITLE
Add (Apple Intelligence) image playground to the Quick Editor

### LINKED ISSUE #568: Create and upload an image through Playground AI
Create and upload an image through Playground AI
Uploading an avatar from Playground is accessible in the Upload menu. Copy: “Playground”

### PR DESCRIPTION
**Pull request by pinarol**
Closes https://github.com/Automattic/Gravatar-SDK-iOS/issues/568#issue-2675522828

### Description

Adds the image playground of Apple to the QuickEditor.

<tr>
    <td></td>
    <td></td>
     <td></td>
  </tr>
</table>

https://github.com/user-attachments/assets/c03d8ab4-8ac2-4ba7-9b8d-4f29b221e416

### Testing Steps

Upgrade to iOS 18.2 beta
Turn on Apple Intelligence
Go to the demo app > Quick Editor flow
Choose "Playground"
- [x] Finish the flow
- [x] Cancel the flow

Test other image sources as well
- [x] Camera
- [x] Image picker.

Turn off Apple Intelligence via device settings
- [x] "Playground" option shouldn't be there anymore

### CODE REVIEW

File: Demo/Gravatar-Demo.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -498,6 +501,7 @@
 				ENABLE_USER_SCRIPT_SANDBOXING = NO;
 				GENERATE_INFOPLIST_FILE = YES;
 				INFOPLIST_FILE = "Demo/Gravatar-Demo/Info.plist";
+				INFOPLIST_KEY_NSCameraUsageDescription = "Accessing camera to create an avatar.";
```
Comment (pinarol) [inline Demo/Gravatar-Demo.xcodeproj/project.pbxproj:504]: I guess `NSCameraUsageDescription` was removed during the merge of two demo apps. So I am adding it back.

File: Sources/GravatarUI/SwiftUI/AvatarPicker/SystemImagePicker/SystemImagePickerView.swift [hunk 2]
```diff
@@ -15,27 +16,37 @@ private struct ImagePicker<Label, ImageEditor: ImageEditorView>: View where Labe
...
+            }
+            return cases
+        }
 
         var id: Int {
             self.hashValue
         }
     }
 
-    @State var isPresented = false
```
Comment (pinarol) [inline Sources/GravatarUI/SwiftUI/AvatarPicker/SystemImagePicker/SystemImagePickerView.swift:24]: This isn't used from anywhere so I am removing it.

File: Sources/GravatarUI/SwiftUI/AvatarPicker/SystemImagePicker/SystemImagePickerView.swift [hunk 3]
```diff
@@ -44,29 +55,58 @@ private struct ImagePicker<Label, ImageEditor: ImageEditorView>: View where Labe
...
-                        } onCancel: {
-                            imagePickerSelectedItem = nil
-                        }.ignoresSafeArea()
-                    }
-                })
+        .imagePlaygroundSheetIfAvailable(
+            isPresented: Binding(
+                get: { sourceType == .playground },
+                set: { if !$0 { sourceType = nil } }
+            ),
```
Comment (pinarol) [inline Sources/GravatarUI/SwiftUI/AvatarPicker/SystemImagePicker/SystemImagePickerView.swift:62]: Doing this to be able to add the new `case playground` to `SourceType` and still have these be managed by a single `sourceType` variable.

(The system offers a modifier for the playground not a View. But our existing image sources are Views. So they are displayed with different methods.)

File: Sources/GravatarUI/SwiftUI/AvatarPicker/SystemImagePicker/SystemImagePickerView.swift [hunk 4]
```diff
@@ -15,27 +16,37 @@ private struct ImagePicker<Label, ImageEditor: ImageEditorView>: View where Labe
     enum SourceType: CaseIterable, Identifiable {
         case photoLibrary
         case camera
+        case playground
+
+        static var allCases: [SourceType] {
+            var cases: [SourceType] = [.camera, .photoLibrary]
+            if #available(iOS 18.2, *) {
+                if EnvironmentValues().supportsImagePlayground {
```
Comment (andrewdmontgomery) [inline Sources/GravatarUI/SwiftUI/AvatarPicker/SystemImagePicker/SystemImagePickerView.swift:24]: Works flawlessly

**Review by andrewdmontgomery — APPROVED** [review]
Worked great!  A few notes:

Images generated in this Playground sheet are available in my Playground app.  That's a nice and unexpected feature of the service.

I had something unexpected happen.  It's probably an issue with the beta. but I'll mention it anyway:

1. Performed testing steps, as instructed
2. After disabling Apple Intelligence, and testing in that state, I re-enabled Apple Intelligence
3. I put the phone down and started looking at the PR diff
4. When I wasn't looking, the phone restarted (crash?)... I had to re-enter my pin in order  to unlock the iphone
5. I went back to the Demo app, and immediately tried to use the playground again.
6. When the Playground sheet opened, I saw the message below
7. I launched the Playground app to see if it gave the same message, but was fully functional
8. Returned to the Demo app, and the Playground worked

Did I just have remarkable timing, and the image playground models finished downloading in the second it took me to launch the Playground app?

The crash is very suspicious, of course.  So I'm attributing this to "beta".

IMG_4559

**Comment by pinarol** [comment]
Yeah that sounds like a system issue...

**Pull request merged**